src/observers/errorMessageObserver.ts

```diff
@@ -1,17 +1,25 @@
 import * as fs from 'fs-extra';
 import * as vscode from 'vscode';
 import { EventType } from '../common/eventType';
-import { BaseEvent, UserSignInFailed, BuildTriggerFailed, RepositoryEnabledV3 } from '../common/loggingEvents';
+import { BaseEvent, UserSignInFailed, BuildTriggerFailed, RepositoryEnabledV3, UserSignInCompleted, SignResult, UserSignOutCompleted, UserSignOutFailed } from '../common/loggingEvents';
 import { MessageAction } from '../shared';
 import { TriggerErrorType } from '../build/triggerErrorType';
 import { safelyReadJsonFile } from '../utils/utils';
 
 export class ErrorMessageObserver {
     public eventHandler = (event: BaseEvent) => {
         switch (event.type) {
-            case EventType.UserSignInFailed:
-                let asUserSignInFailed = <UserSignInFailed>event;
-                this.showErrorMessage(`Sign-In failed: ${asUserSignInFailed.message}`);
+            case EventType.UserSignInCompleted:
+                if ((<UserSignInCompleted>event).result === SignResult.Failed) {
+                    let asUserSignInFailed = <UserSignInFailed>event;
+                    this.showErrorMessage(`Sign-In failed: ${asUserSignInFailed.err.message}`);
+                }
+                break;
+            case EventType.UserSignOutCompleted:
+                if ((<UserSignOutCompleted>event).result === SignResult.Failed) {
+                    let asUserSignOutFailed = <UserSignOutFailed>event;

```

Comment:
there was `UserSignOutFailed ` and `UserSignOutSucceeded `, the reason why I combine them into one is I hope the count of eventType can be as  less as possible, which can make the system less complex.
and another reason is this can be more consistent with the telemetry event type.

---
In reply to: [376908723](https://github.com/docascode/vscode-docs-build/pull/35#discussion_r376908723) [](ancestors = 376908723)

Reply:
IMHO, it's more complex to have nested types.
But if this is for consistency with telemetry events, then it's fine